pkg/eventcounter/api/api_test.go

```diff
@@ -919,6 +921,200 @@ func TestGenInterval(t *testing.T) {
 	}
 }
 
+func TestGetOneMonthTimeStamps(t *testing.T) {
+	t.Parallel()
+
+	endAt := time.Now()
+	startAt, err := genInterval(jpLocation, endAt, 30)
+	assert.NoError(t, err)
+
+	patterns := []struct {
+		desc             string
+		startAt          time.Time
+		expectedElements []int64
+		expectedLen      int
+	}{
+		{
+			desc:    "success",
+			startAt: startAt,
+			expectedElements: []int64{
+				getDate(startAt),
+				getDate(startAt.AddDate(0, 0, 1)),
+				getDate(endAt),
+			},
+			expectedLen: 31,
+		},
+	}
+	for _, p := range patterns {
+		t.Run(p.desc, func(t *testing.T) {
+			actual := getOneMonthTimeStamps(p.startAt)
+			for _, e := range p.expectedElements {
+				assert.Contains(t, actual, int64(e))
+			}
+			assert.Len(t, actual, p.expectedLen)
+		})
+	}
+}
+
+func TestGetVariationIDs(t *testing.T) {
+	t.Parallel()
+
+	vID1 := newUUID(t)
+	vID2 := newUUID(t)
+
+	patterns := []struct {
+		desc       string
+		variations []*featureproto.Variation
+		expected   []string
+	}{
+		{
+			desc: "success",
+			variations: []*featureproto.Variation{
+				{
+					Id:    vID1,
+					Value: "true",
+				},
+				{
+					Id:    vID2,
+					Value: "false",
+				},
+			},
+			expected: []string{
+				vID1, vID2, defaultVariationID,
+			},
+		},
+	}
+	for _, p := range patterns {
+		t.Run(p.desc, func(t *testing.T) {
+			actual := getVariationIDs(p.variations)
+			assert.Equal(t, actual, p.expected)
+		})
+	}
+}
+
+func TestGetEvaluationTimeseriesCountV2(t *testing.T) {
+	t.Parallel()
+	mockController := gomock.NewController(t)
+	defer mockController.Finish()
+
+	patterns := []struct {
```

Comment:
Let's also add an internal error pattern for each mock interface to ensure the expected error is correct.